src/lib/OpenEXR/ImfDeepScanLineInputFile.cpp

```diff
@@ -9,2357 +9,1021 @@
-
-    const char* readPtr;
-
-    //
-    // If the sample count table is compressed, we'll uncompress it.
-    //
+#if ILMTHREAD_THREADING_ENABLED
+    int64_t nchunks;
+    nchunks = ((int64_t) scanLine2 - (int64_t) scanLine1);
+    nchunks /= (int64_t) scansperchunk;
+    nchunks += 1;
 
-    if (sampleCountTableDataSize <
-        static_cast<uint64_t> (data->maxSampleCountTableSize))
+    if (nchunks > 1 && numThreads > 1)
     {
-        if (!data->sampleCountTableComp)
-        {
-            THROW (
-                IEX_NAMESPACE::ArgExc,
-                "Deep scanline data corrupt at chunk "
-                    << lineBlockId << " (sampleCountTableDataSize error)");
-        }
-        data->sampleCountTableComp->uncompress (
-            data->sampleCountTableBuffer,
-            static_cast<int> (sampleCountTableDataSize),
-            minY,
-            readPtr);
-    }
-    else
-        readPtr = data->sampleCountTableBuffer;
+        ILMTHREAD_NAMESPACE::TaskGroup tg;
 
-    char* base    = data->sampleCountSliceBase;
-    int   xStride = data->sampleCountXStride;
-    int   yStride = data->sampleCountYStride;
+        for (int y = scanLine1; y <= scanLine2; )
+        {
+            if (EXR_ERR_SUCCESS != exr_read_scanline_chunk_info (*_ctxt, partNumber, y, &cinfo))
+                throw IEX_NAMESPACE::InputExc ("Unable to query scanline information");
 
-    // total number of samples in block: used to check samplecount table doesn't
-    // reference more data than exists
+            // used for honoring the numThreads
+            _sem.wait ();
 
-    size_t cumulative_total_samples = 0;
+            ILMTHREAD_NAMESPACE::ThreadPool::addGlobalTask (
+                new LineBufferTask (&tg, this, &fb, cinfo, y, scanLine2, countsOnly) );
 
-    for (int y = minY; y <= maxY; y++)
+            y += scansperchunk - (y - cinfo.start_y);
+        }
+    }
+    else
+#endif
     {
-        int yInDataWindow                    = y - data->minY;
-        data->lineSampleCount[yInDataWindow] = 0;
+        auto sp = getChunkProcess ();
+        bool redo = sp->first || sp->counts_only != countsOnly;
 
-        int lastAccumulatedCount = 0;
-        for (int x = data->minX; x <= data->maxX; x++)
+        sp->counts_only = countsOnly;
+        for (int y = scanLine1; y <= scanLine2; )
         {
-            int accumulatedCount, count;
-
-            //
-            // Read the sample count for pixel (x, y).
-            //
-
-            Xdr::read<CharPtrIO> (readPtr, accumulatedCount);
-
-            // sample count table should always contain monotonically
-            // increasing values.
-            if (accumulatedCount < lastAccumulatedCount)
+            if (EXR_ERR_SUCCESS != exr_read_scanline_chunk_info (*_ctxt, partNumber, y, &cinfo))
+                throw IEX_NAMESPACE::InputExc ("Unable to query scanline information");
+
+            // do we have the same chunk where we can just re-run the unpack
```

Comment:
Could you use "check if we have" wording? I had to think about the comment, because sometimes in your comments, sentences starting with "do we" mean code you plan to revisit, but I think in this case it's fine